src/hooks/use-load-data-pages.ts

```diff
@@ -35,29 +59,34 @@ const useLoadDataPages = (model: Model, layoutService: LayoutService, viewServic
   usePromise(async () => {
```

Comment:
You need to first check if it is a snapshot and if so use the data page from that.

Example:
```js
usePromise(async () => {
    if (layoutService.isSnapshot) {
      setDataPages(layoutService.layout.snapshotData?.content?.qPivotDataPages || []);
    } else if (
      model &&
      (shouldFetchAdditionalData(qLastExpandedPos, viewService) || isMissingLayoutData(layoutService))
    ) {
      try {
        const pivotPages = await (model as EngineAPI.IGenericObject).getHyperCubePivotData(Q_PATH, [
          {
            qLeft: qLastExpandedPos ? viewService.gridColumnStartIndex : 0,
            qTop: qLastExpandedPos ? viewService.gridRowStartIndex : 0,
            qWidth: !viewService.gridWidth ? DEFAULT_PAGE_SIZE : viewService.gridWidth,
            qHeight: !viewService.gridHeight ? DEFAULT_PAGE_SIZE : viewService.gridHeight,
          },
        ]);

        setDataPages([...qHyperCube.qPivotDataPages, ...pivotPages]);
      } catch (error) {
        // TODO handle error
        console.error(error);
      }
    } else if (qHyperCube.qPivotDataPages.length) {
      // when we get new page
      setDataPages(qHyperCube.qPivotDataPages);
    }

    ref.isLoading = false;
  }, [layoutService.layout, model, viewService, snapshotData]);
```

Reply:
Note that I've updated the example so it should now work with rendering tests